tests/unit/services/importing/aioseo-robots-service-test.php

```diff
@@ -0,0 +1,333 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Unit\Services\Importing;
+
+use Mockery;
+use Brain\Monkey;
+use Yoast\WP\SEO\Services\Importing\Aioseo_Robots_Service;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+use Yoast\WP\SEO\Tests\Unit\Doubles\Services\Importing\Aioseo_Robots_Service_Double;
+
+/**
+ * Class Aioseo_Robots_Service_Test.
+ *
+ * @group importing
+ *
+ * @coversDefaultClass \Yoast\WP\SEO\Services\Importing\Aioseo_Robots_Service
+ * @phpcs:disable Yoast.Yoast.AlternativeFunctions.json_encode_json_encode
+ */
+class Aioseo_Robots_Service_Test extends TestCase {
+
+	/**
+	 * Represents the mock instance to test.
+	 *
+	 * @var Mockery\MockInterface|Aioseo_Robots_Service
+	 */
+	protected $mock_instance;
+
+	/**
+	 * The class under test.
+	 *
+	 * @var Aioseo_Robots_Service
+	 */
+	protected $aioseo_robots_service;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function set_up() {
+		$this->aioseo_robots_service = new Aioseo_Robots_Service();
+		$this->mock_instance         = Mockery::mock( Aioseo_Robots_Service_Double::class )->makePartial()->shouldAllowMockingProtectedMethods();
```

Comment:
[Please don't introduce new partial mocks](https://yoast.atlassian.net/wiki/spaces/DEV/pages/1081770041/Architecture+decision+log#Partial-mocks-are-discouraged-and-should-never-be-used-for-dependencies.)

It feels like this robots_service could be split up in a `AIOSEO_(Robot_)Settings_Provider` and a `AOISEO_Robots_Transformer`

Reply:
Yeap, my bad ✋ 

I splt the robot service now